src/OrleansRuntime/Messaging/Gateway.cs

```diff
@@ -431,6 +431,8 @@ protected override void Process(OutgoingClientMessage request)
                 {
                     if (Log.IsVerbose3) Log.Verbose3("Sent message {0} to client {1}", msg, client);
                 }
+
+                msg.Dispose();
```

Comment:
I think we only want to dispose here if send was successful, if send fails and we reenque into pending sends, this dispose would be bad, right?